### DISCUSSION TITLE
Extract drawings data to png

### DISCUSSION THREAD

**Discussion by daoth90** [pymupdf/PyMuPDF#1032] [Q&A]
Hi the community, last time I was able to extract all drawings data to a new pdf file but my goal is able to extract the image created by drawings data to png.

Please give me some suggestions.

**Comment by JorjMcKie** [comment]
When you re-draw the drawings extracted by `page.get_drawings()` on a new page, then the `Shape` object you used will "log" a rectangle for you, which gets updated with each draw method.
You can use this rectangle as the `clip` parameter when making a pixmap of the new page. This pixmap can be save as a PNG obviously.

**Reply by daoth90** [reply]
If a page contains 3 images and page.get_drawings () return array 100 items, how do I take paths of each image to extract them into separate images?

**Reply by JorjMcKie** [reply]
Each item of the list `page.get_drawings()` is a "path" in PDF terminology. It may be (and often is) as simple ("atomic") as one line, or one rectangle. but occasionally also quite complex and consist of many / hundreds of single lines, curves or rectangles.
Each of these items still has a `rect` property, which is the rectangle wrapping the single atomic draws.

When you re-create a path on a new page you would iterate through `path["items"]`, which is the list of such "atomic" draw commands, using `shape.draw_*` on the target page.

To decide, how to group your 100 paths into meaningful few images on the target, is left to your wits. You need some heuristics here. You would probably try to "group" or cluster the rectangles somehow ... look at wether some are contained in others or have non-empty overlaps ...
Not easy for sure.
Start with one rectangle, look for another with a non-empty overlap, build their union rectangle, take that union rectangle and check again with the rest of the draw rects, etc.
When no more intersecting rect found, pick another rectangle from the remaining draw rectangles, and so on.
Just an idea. Don't bully me if something goes wrong 😎

**Comment by JorjMcKie** [comment]
Don't know if still relevant: I have experimented a little and found this script quite useful: 
[path2png.zip](https://github.com/pymupdf/PyMuPDF/files/6631931/path2png.zip)

**Reply by rafayaar** [reply]
@JorjMcKie Attached script is I guess redundant. Width is always None for PDFs when getting drawings. If available please provide updated code
